src/base/bittorrent/torrenthandleimpl.cpp

```diff
@@ -851,15 +828,21 @@ TorrentState TorrentHandleImpl::state() const
 
 void TorrentHandleImpl::updateState()
 {
-    if (hasError()) {
-        m_state = TorrentState::Error;
-    }
-    else if (m_nativeStatus.state == lt::torrent_status::checking_resume_data) {
+    if (m_nativeStatus.state == lt::torrent_status::checking_resume_data) {
```

Comment:
if you want, you can use a switch():
```c++
switch (m_nativeStatus.state) {
case lt::torrent_status::checking_resume_data:
    m_state = TorrentState::CheckingResumeData;
    return;
//...
}
if (isMoveInProgress())
// ...
```

Reply:
I don't feel that it's better (especially the need for these `return` statements).